extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/SingleValueScrutinizer.java

```diff
@@ -44,6 +46,17 @@
 public class SingleValueScrutinizer extends EditScrutinizer {
 
     public static final String type = "single-valued-property-added-more-than-once";
+    public static String SINGLE_VALUE_CONSTRAINT_QID = "Q19474404";
+    public static String SINGLE_BEST_VALUE_CONSTRAINT_QID = "Q52060874";
+
+    class SingleValueConstraint extends Constraint {
+        Set<EntityIdValue> constraintExceptions;
+
+        SingleValueConstraint(Statement statement) {
+            super(statement);
+            constraintExceptions = getConstraintExceptions();
```

Comment:
You should not need to declare a new field for constraint exceptions again: you can just reuse the one from the super class. So, for all constraints which do not take any parameters, you should not need to declare any inner class: the `Constraint` class itself can be used in the scrutinizer.